### PR TITLE
{AKS} Update tests to use vnet integration

### PR DESCRIPTION
**Pull request by lzhecheng**
private keyvault + konnectivity is blocked now, so vnet integration should be used.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command

az aks create/update --enable-azure-keyvault-kms

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [ ] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
KMS

**Review by FumingZhang — COMMENTED** [review]
Queued [live test](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=104076711&view=results) to validate the change

- test_aks_create_with_azurekeyvaultkms_private_key_vault
- test_aks_update_with_azurekeyvaultkms_private_key_vault
- test_aks_create_with_azurekeyvaultkms_private_cluster_v1_private_key_vault

File: src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py [hunk 1]
```diff
@@ -9220,9 +9220,11 @@ def test_aks_create_with_azurekeyvaultkms_private_key_vault(
 
         create_cmd = (
             "aks create --resource-group={resource_group} --name={name} "
+            "--aks-custom-headers AKSHTTPCustomFeatures=Microsoft.ContainerService/EnableAPIServerVnetIntegrationPreview "
             "--assign-identity {identity_id} "
             "--enable-azure-keyvault-kms --azure-keyvault-kms-key-id={key_id} "
             "--azure-keyvault-kms-key-vault-network-access=Private --azure-keyvault-kms-key-vault-resource-id {kv_resource_id} "
+            "--enableVnetIntegration=true "
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py:9227]: The option name should be `--enable-apiserver-vnet-integration`

**Review by FumingZhang — COMMENTED** [review]
Please use {AKS} as the beginning of the PR title

File: src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py (hunk 1 above)
Comment (lzhecheng) [reply src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py]: Good catch!

**Comment by FumingZhang** [comment]
> Queued [live test](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=104076711&view=results) to validate the change
> 
> * test_aks_create_with_azurekeyvaultkms_private_key_vault
> * test_aks_update_with_azurekeyvaultkms_private_key_vault
> * test_aks_create_with_azurekeyvaultkms_private_cluster_v1_private_key_vault

[Requeued](https://msazure.visualstudio.com/CloudNativeCompute/_build/results?buildId=104197897&view=results), test passed!

**Review by FumingZhang — APPROVED** [review]
LGTM

**Comment by lzhecheng** [comment]
@zhoxing-ms @yanzhudd Could you please merge this one? Thanks.

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**